src/slic3r/GUI/OG_CustomCtrl.cpp

```diff
@@ -373,22 +373,31 @@ void OG_CustomCtrl::OnMotion(wxMouseEvent& event)
             break;
         }
 
-        for (size_t opt_idx = 0; opt_idx < line.rects_undo_icon.size(); opt_idx++)
+        size_t undo_icons_cnt = line.rects_undo_icon.size();
```

Comment:
Curiously, what are these changes for?

Reply:
I believe a lot of it is a refactor. Rather than having multiple loops, it is a single loop that handles all cases